### PR TITLE
Add _val_or_rc-function

### PR DESCRIPTION
**Pull request by oscargus**
## PR summary

``` python
if foo is None:
    foo = mpl.rcParams["bar.foo"]
```
is a very common code pattern, so may be good to have a helper for it. ~Right now I only considered `legend.py`, but if this is an accepted idea there are many places where it can be used.~

(And, yes, the reason I started with legend is that it was defined as a local function there.)

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

**Review by efiring — CHANGES_REQUESTED** [review]
Because `legend.val_or_rc` is formally public API, it probably should be removed after a standard deprecation cycle.

**Comment by oscargus** [comment]
I do not think that? I think it is an internal function in `__init__` that is not accessible from the outside?

**Review by efiring — APPROVED** [review]
Oops! You are right, of course.   All set.

File: lib/matplotlib/legend.py [hunk 1]
```diff
@@ -603,11 +592,9 @@ def val_or_rc(val, rc_name):
             'markeredgecolor': ['get_markeredgecolor', 'get_edgecolor'],
             'mec':             ['get_markeredgecolor', 'get_edgecolor'],
         }
+        labelcolor = mpl._val_or_rc(labelcolor, 'legend.labelcolor')
```
Comment (greglucas) [inline lib/matplotlib/legend.py:595]: You could nest another `mpl._val_or_rc(..., 'text.color')` here and add a short comment about falling back to `text.color` if `legend.labelcolor` is not present.

File: lib/matplotlib/legend.py [hunk 2]
```diff
@@ -447,20 +443,17 @@ def __init__(
         #: instance.
         self._custom_handler_map = handler_map
 
-        def val_or_rc(val, rc_name):
-            return val if val is not None else mpl.rcParams[rc_name]
```
Comment (greglucas) [inline lib/matplotlib/legend.py:451]: I think you could possibly save a bunch of method gets below by setting this and reusing it?
`val_or_rc = mpl._val_or_rc`

File: lib/matplotlib/__init__.py [hunk 3]
```diff
@@ -1294,6 +1294,13 @@ def is_interactive():
     return rcParams['interactive']
 
 
+def _val_or_rc(val, rc_name):
+    """
+    If *val* is None, return ``mpl.rcParams[rc_name]``, otherwise return val.
```
Comment (greglucas) [inline lib/matplotlib/__init__.py:1299]:
```suggestion
    Return the value if present, otherwise lookup `rc_name` in the rcParams dictionary.
```
I'd suggest rewording this to be more human friendly since the code already explains what you had written.

File: lib/matplotlib/legend.py (hunk 2 above)
Comment (timhoffm) [reply lib/matplotlib/legend.py:451]: I would like to warn on ineffective micro optimizations. Squeezing out nanoseconds does not make any measureable difference.

A function lookup cost 30ns (Python 3.11):
```
In [2]: %timeit mpl.rc_params_from_file
29.7 ns ± 0.544 ns per loop
```

If we want to improve performance (and there's a lot to gain) that would need to be guided by profiling to identify the real pain points. In other cases, code structure/readability should be the primary guideline. - Not shure what is better here, but I tend to not do the additional logic indirection for saving a method to a new variable just to save method lookups.

File: lib/matplotlib/__init__.py (hunk 3 above)
Comment (timhoffm) [reply lib/matplotlib/__init__.py:1299]: Using None and ``mpl.rcParams[rc_name]`` is good, because that's precise and comprehensible. Optional, slightly less algorithmic variant.

```suggestion
    Pass *val* through unless it is None, in which case return ``mpl.rcParams[rc_name]``.
```

**Review by timhoffm — APPROVED** [review]
